z_calibration.py

```diff
@@ -108,11 +108,6 @@ def handle_connect(self):
             mesh = self.printer.lookup_object('bed_mesh', default=None)
             if mesh is None or mesh.bmc.relative_reference_index is None:
```

Comment:
If this can now be filled out at runtime I probably should not get an error here if I don't have `relative_reference_index` configured in my default bed mesh. I think we could delete all the way up to line 105.

Reply:
I wonder what happens then, if you just run BED_MESH_CALIBRATE without parameters and you don't have a RRI defined in the config? I think we should force the users to define a RRI when using z calibration.

Reply:
Not necessarily. If we run a BED_MESH_CALIBRATE for the sub-area of the bed the RRI can be passed at that time, even if this value is not defined. You are forcing it to be defined in the config but the value isn't actually read until later. Do the checking when the value is required.